src/Items/ItemLilypad.h

```diff
@@ -100,6 +118,13 @@ class cItemLilypadHandler:
 
```

Comment:
Ah, this is kinda embarrassing on my part. The original commit for lilypads was 283a66bcae5f26d2b5038ba0959314a687c4d8b6#diff-c25a12152914ebf4cd50f2ac9cd08ce8631772948705300e963bc97d96c31b30. I then proceeded to introduce a regression in f763242e5c5d8e6469685c4be95d3dc264ed6a67#diff-c25a12152914ebf4cd50f2ac9cd08ce8631772948705300e963bc97d96c31b30 by deleting the GetBlock with the new coordinates. You can see the remains of that check on line 96 above. Could you move the check from below into the callback?

Reply:
To do this I would need to re-introduce the `cWorld * m_World` variable into the callback. Right now it doesn't seem like the callback has a way to run `GetBlock`.

Should I just re-add it?

Reply:
True, in that case we should delete the redundant check in the callback altogether.

Reply:
Gotcha, I just tested with the check in the callback and it seems to have some problems anyways
- If you're completely submerged in water then you can still place a lilypad due to `!IsBlockWater(Block)`
- If there's a half slab above the water (doesn't occupy a full voxel) then the half slab gets replaced due to `cBlockInfo::FullyOccupiesVoxel(Block)`